src/compiler/program.ts

```diff
@@ -2894,10 +2894,11 @@ namespace ts {
                 path += (i === 2 ? "/" : "-") + components[i];
                 i++;
             }
-            // The strange "/fake.file" is to ensure we first resolve in node_modules for the current directory.
-            const localOverride = resolveModuleName("@typescript/" + path, currentDirectory + "/fake.file", { moduleResolution: ModuleResolutionKind.NodeJs }, host, moduleResolutionCache);
-            if (localOverride?.resolvedModule) {
-                return localOverride.resolvedModule.resolvedFileName;
+
+            const resolveFrom = combinePaths(currentDirectory, `/__lib_node_modules_lookup_${libFileName}__.ts`);

```

Comment:
This always ignores `currentDirectory` because the latter argument is absolute.

Reply:
Guessing this didn’t break tests only because all the tests are rooted at `/`

Reply:
nice spot!